opennav_following/include/opennav_following/following_server.hpp

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) 2024 Open Navigation LLC
+// Copyright (c) 2024 Alberto J. Tudela Roldán
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef OPENNAV_FOLLOWING__FOLLOWING_SERVER_HPP_
+#define OPENNAV_FOLLOWING__FOLLOWING_SERVER_HPP_
+
+#include <vector>
+#include <memory>
+#include <string>
+#include <mutex>
+#include <functional>
+
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_util/lifecycle_node.hpp"
+#include "nav2_util/node_utils.hpp"
+#include "nav2_util/simple_action_server.hpp"
+#include "nav2_util/twist_publisher.hpp"
+#include "opennav_docking/controller.hpp"
+#include "opennav_docking/pose_filter.hpp"
+#include "opennav_following_msgs/action/follow_object.hpp"
+#include "tf2_ros/buffer.h"
+#include "tf2_ros/transform_listener.h"
+
+namespace opennav_following
+{
+/**
+ * @class opennav_following::FollowingServer
+ * @brief An action server which implements a dynamic following behavior
+ */
+class FollowingServer : public nav2_util::LifecycleNode
```

Comment:
There seems to be a great deal of duplication. Would it be possible or sensible to have this be a node that derives from the docking server and/or find a way to have it share more common code?